metadata-patch.md

```diff
@@ -0,0 +1,195 @@
+# Modifying metadata via Patch
+[Back to the index](README.md)
+
+**Contents:**
+
+* [Introduction](#introduction)
+* [Adding metadata](#adding-metadata)
+* [Removing metadata](#removing-metadata)
+* [Replacing metadata](#replacing-metadata)
+* [Ordering metadata](#ordering-metadata)
+
+## Introduction
+
+A user with write permission on a DSpace object may change any of its metadata
+using an HTTP PATCH request. All DSpace object types (Bitstreams, Items, Collections, etc.)
+support metadata changes in roughly the same way.
+
+This guide is focused on metadata-only changes and omits non-metadata parts of the object's
+json representation for brevity. However, you should be aware that PATCH requests operate
+against the whole object, and may therefore modify other mutable attributes in the same
+request.
+
+For additional information about PATCH support in DSpace, including request format and response
+status code details, see [General rules for the Patch operation](patch.md)
+
+The examples in each section below build on each other, assuming an initial metadata state of:
+
+```json
+{
+  "metadata": {
+    "dc.title": [
+      { "value": "Initial Title", "language": null, "authority": null, "confidence": -1 }
+    ]
+  }
+}
+```
+
+## Adding metadata
+
+With the `add` operation, you can add any number of metadata values in a single request.
+
+Note:
+
+* When no values exist yet for a metadata key, you must add the key and value array with one
+  or more values.
```

Comment:
the tricky part to understand here is that you cannot use `/metadata/dc.title/-` to set the initial title but instead you need to initialize the metadata using the path `/metadata/dc.title`

This is explained with different wording here
https://github.com/DSpace/Rest7Contract/blob/master/workspaceitem-data-metadata.md

> Please note that according to the JSON Patch specification RFC6902 to initialize a new metadata in the section the add operation must receive an array of values and it is not possible to add a single value to the not yet initialized "/sections/<:name-of-the-form>/<:metadata>/-" path.

I leave this comment only to help other reviewer to suggest, if necessary, a better wording or uniformity in our doc

Reply:
I like the idea of borrowing language here from the existing documentation (as @abollini suggests), and also provide an example of adding an *initial* metadata value.  That will make this documentation clearer.

Is there an opportunity to somehow *consolidate* these two areas of the REST Contract?  Could we think about even having the "workspaceitem-data-metadata.md" *link* to the more generic "metadata-patch.md"?  I don't know that we need to hold up this PR for that consolidation, but it seems like there are very similar concepts/examples that will exist in both of these areas of the REST Contract.

Reply:
I made the language consistent on this topic, and cross-linked both documents in the latest commit.